src/content/developers/tutorials/hello-world-smart-contract-fullstack/index.md

```diff
@@ -0,0 +1,1548 @@
+   constructor(string memory initMessage) {
+
+      // Accepts a string argument `initMessage` and sets the value into the contract's `message` storage variable).
+      message = initMessage;
+   }
+
+   // A public function that accepts a string argument and updates the `message` storage variable.
+   function update(string memory newMessage) public {
+      string memory oldMsg = message;
+      message = newMessage;
+      emit UpdatedMessages(oldMsg, newMessage);
+   }
+}
+```
+
+This is a super simple smart contract that stores a message upon creation and can be updated by calling the `update` function.
+
+### Step 11: Connect Metamask & Alchemy to your project {#step-11-connect-metamask-&-alchemy-to-your-project}
+
+We’ve created a Metamask wallet, Alchemy account, and written our smart contract, now it’s time to connect the three.
+
+Every transaction sent from your virtual wallet requires a signature using your unique private key. To provide our program with this permission, we can safely store our private key (and Alchemy API key) in an environment file.
+
+> To learn more about sending transactions, check out [this tutorial](https://docs.alchemyapi.io/alchemy/tutorials/sending-transactions-using-web3-and-alchemy) on sending transactions using web3.
+
+First, install the dotenv package in your project directory:
+
+```
+npm install dotenv --save
+```
+
+Then, create a `.env` file in the root directory of our project, and add your Metamask private key and HTTP Alchemy API URL to it.
+
+Your environment file must be named `.env` or it won't be recognized as an environment file.
+
+Do not name it `process.env` or `.env-custom` or anything else.
+
+- Follow [these instructions](https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key) to export your private key
+- See below to get HTTP Alchemy API URL
+
+![](./get-alchemy-api-key.gif)
+
+Your `.env` should look like this:
+
+```
+API_URL = "https://eth-ropsten.alchemyapi.io/v2/your-api-key"
+PRIVATE_KEY = "your-metamask-private-key"
+```
+
+To actually connect these to our code, we’ll reference these variables in our `hardhat.config.js` file on step 13.
+
+### Step 12: Install Ethers.js {#step-12-install-ethers.js}
+
+Ethers.js is a library that makes it easier to interact and make requests to Ethereum by wrapping [standard JSON-RPC methods](https://docs.alchemyapi.io/alchemy/documentation/alchemy-api-reference/json-rpc) with more user friendly methods.
+
+Hardhat makes it super easy to integrate [Plugins](https://hardhat.org/plugins/) for additional tooling and extended functionality. We’ll be taking advantage of the [Ethers plugin](https://hardhat.org/plugins/nomiclabs-hardhat-ethers.html) for contract deployment ([Ethers.js](https://github.com/ethers-io/ethers.js/) has some super clean contract deployment methods).
+
+In your project directory type:
+
+```bash
+npm install --save-dev @nomiclabs/hardhat-ethers "ethers@^5.0.0"
```

Comment:
@corwintines was there a windows vs unix issue here? Did we find a way to resolve this?